### PR TITLE
One group form (merge start, edit forms into one)

### PR DESCRIPTION
**Pull request by robguthrie**

### CODE REVIEW

**Comment by robguthrie** [comment]
@rdbartlett I've dropped all the pricing messaging.. which is only because when we talked about it at the other workshop ben expressed he thought "choose a payment plan" made it sound like free is not an option. Can we jam on the cool ultimate language tomorrow morning?

Start group:
image
Edit group:
image

Start subgroup:
image

Edit subgroup:
image

**Comment by robguthrie** [comment]
also @rdbartlett would be great to jam on language for: subgroup form should have line of text explaining what a subgroup is for.

**Comment by rdbartlett** [comment]
Yeeha looks great

**Comment by gdpelican** [comment]
This is going to trample the soft drafts stuff for group, unfortunately.

**Comment by robguthrie** [comment]
Sorry about that! In what parts is it clashing?

File: lineman/app/components/group_form/group_form.coffee [hunk 1]
```diff
@@ -0,0 +1,65 @@
+angular.module('loomioApp').factory 'GroupForm', ->
+  templateUrl: 'generated/components/group_form/group_form.html'
+  controller: ($scope, $rootScope, $location, group, FormService, Records, $translate, PrivacyString) ->
+    $scope.group = group.clone()
+
+    $scope.$on 'modal.closing', (event) ->
+      FormService.confirmDiscardChanges(event, $scope.group)
+
+    $scope.i18n = do ->
+      h = {}
```
Comment (gdpelican) [inline lineman/app/components/group_form/group_form.coffee]: What is 'h'? And I kinda find that these deep if / else logic structures are really difficult to follow; I'd much rather have some sort of separate service or method that would determine each value on it's own, based on the group

```
.group-heading{translate: '{{groupHeadingFor(group)}}'}
.group-name{translate: '{{groupNameFor(group)}}'}
.group-submit{translate: '{{groupSubmitFor(group)}}'}
```

File: lineman/app/components/group_form/group_form.coffee [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    $scope.expandForm = ->
+      $scope.expanded = true
+
+    $scope.privacyStatement = ->
+      PrivacyString.groupPrivacyStatement($scope.group)
+
+    $scope.privacyStringFor = (state) ->
+      PrivacyString.group($scope.group, state)
+
+    $scope.buh = {}
```
Comment (gdpelican) [inline lineman/app/components/group_form/group_form.coffee]: What is 'buh' ??

File: lineman/app/components/group_page/gift_card/gift_card.coffee [hunk 3]
```diff
@@ -5,6 +5,11 @@ angular.module('loomioApp').directive 'giftCard', ->
   replace: true
   controller: ($scope, $window, AppConfig, CurrentUser) ->
 
+    $scope.show = ->
+      CurrentUser.isMemberOf($scope.group) and
```
Comment (gdpelican) [inline lineman/app/components/group_page/gift_card/gift_card.coffee]: This should be wrapped into the AbilityService

File: lineman/app/components/group_page/trial_card/trial_card.coffee [hunk 4]
```diff
@@ -3,7 +3,13 @@ angular.module('loomioApp').directive 'trialCard', ->
   restrict: 'E'
   templateUrl: 'generated/components/group_page/trial_card/trial_card.html'
   replace: true
-  controller: ($scope, ModalService, ChoosePlanModal) ->
+  controller: ($scope, ModalService, ChoosePlanModal, AbilityService, AppConfig,
+               CurrentUser) ->
+
+    $scope.show = ->
+      $scope.group.subscriptionKind == 'trial' and
```
Comment (gdpelican) [inline lineman/app/components/group_page/trial_card/trial_card.coffee]: Same as above; I think this wants to move to AbilityService

File: lineman/app/components/group_form/group_form.coffee (hunk 1 above)
Comment (gdpelican) [inline lineman/app/components/group_form/group_form.coffee]: I wonder if the functionality of 'PrivacyString' could be expanded slightly to include these? Maybe it's actually 'GroupMessaging' or something.

Comment (robguthrie) [inline lineman/app/components/group_form/group_form.coffee]: Yea I agree. Can do. It's a big improvement over having conditions in the template though.

File: lineman/app/components/group_form/group_form.coffee (hunk 2 above)
Comment (robguthrie) [inline lineman/app/components/group_form/group_form.coffee]: Angular work-around. This is one of those cases you can't just put something on the scope due to some dumb angular thing.

**Pull request merged**